src/extension.ts

```diff
@@ -3982,6 +3991,27 @@ const buildFlashAndMonitor = async (runMonitor: boolean = true) => {
       notificationMode === idfConf.NotificationMode.Notifications
         ? vscode.ProgressLocation.Notification
         : vscode.ProgressLocation.Window;
+
+    // This validation doesn't allows users to use build,flash, monitor command if flash encryption is enabled for release mode
+    // because monitoring command resets the device which is not recommended.
+    // Reset should happen by Bootloader itself once it completes encrypting all artifacts.
+    if (isFlashEncryptionEnabled(workspaceRoot)) {
```

Comment:
Could monitor just run without reset by setting `createMonitor(noReset: true)` ?

Reply:
I did not think of this solution, but it would work. I will make the modifications in a new commit

Reply:
I am just waiting for a confirmation from Mahavir

Reply:
I have replaced this message with adding --no-reset flag